### PR TITLE
Warn Windows users when long paths are not enabled

### PR DESCRIPTION
**Pull request by jstvz**
This PR adds a utility function to report whether the environment supports long pathnames and warn the user if they are not. This function is called from `cci version` and `error`.

Sample output:
<details>
<summary>cci version</summary>
image
</details>
<details>
<summary>cci error info</summary>
image
</details>
<details>
<summary>cci error gist</summary>

```
CumulusCI version: 3.53.0.dev1
Python version: 3.10.2 (D:\Users\jestevez\CumulusCI\venv\Scripts\python.exe)
Environment Info: Windows / AMD64
Windows long path support enabled: False
```

</details>

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: cumulusci/cli/utils.py [hunk 1]
```diff
@@ -106,3 +112,29 @@ def check_latest_version():
...
+    )
+
+    is_enabled, _ = winreg.QueryValueEx(access_key, "LongPathsEnabled")
+
+    return is_enabled == 1
+
+
+def warn_if_no_long_paths(console: Console = Console()) -> None:
+    """Print a warning to the user if long paths are not enabled."""
+    print("foo")
```
Comment (davisagli) [inline cumulusci/cli/utils.py:138]: 👀

**Comment by prescod** [comment]
Given the perf cost:

1. Should we run this code once per hour like the version check?
2. Should we run it only until longpaths are enabled? Then assume they will remain enabled?
3. Should we run it 100% of the time in cci error and cci version contexts?

File: cumulusci/cli/utils.py (hunk 1 above)
Comment (jstvz) [reply cumulusci/cli/utils.py:138]: 🥸

File: cumulusci/cli/tests/test_utils.py [hunk 2]
```diff
@@ -65,3 +66,55 @@ def test_check_latest_version_request_error(click, get_latest_final_version):
...
+    not sys.platform.startswith("win"), reason="Requires Windows Registry"
+)
+@mock.patch("winreg.QueryValueEx")
+def test_win32_warning(query_value):
+    query_value.return_value = (1, 1)
+
+    is_enabled = utils.win32_long_paths_enabled()
+
+    query_value.assert_called_once()
+    "LongPathsEnabled" in query_value.cal_args.args
```
Comment (davidmreed) [inline cumulusci/cli/tests/test_utils.py:81]:
```suggestion
    assert "LongPathsEnabled" in query_value.cal_args.args
```

File: cumulusci/cli/tests/test_utils.py [hunk 3]
```diff
@@ -65,3 +66,55 @@ def test_check_latest_version_request_error(click, get_latest_final_version):
...
+
+
+@pytest.mark.skipif(
+    not sys.platform.startswith("win"), reason="Requires Windows Registry"
+)
+@mock.patch("cumulusci.cli.utils.win32_long_paths_enabled")
+@mock.patch("rich.console.Console.print")
+def test_no_longpath_warn_on_win(console_print, is_enabled):
+    is_enabled.return_value = True
+
```
Comment (davidmreed) [inline cumulusci/cli/tests/test_utils.py:104]: Missing call to method under test?

**Review by davidmreed — CHANGES_REQUESTED** [review]
Looks great - just flagging a couple lines in tests that look like they're false-positiving.

**Review by davidmreed — APPROVED** [review]

**Pull request merged**